Sentinel/Classes/Core/Internal/SentinelInternal.swift

```diff
@@ -26,37 +28,75 @@ extension Sentinel {
-
-        let toolItems = tools.filter { tool in
-            return tool.type != .application &&
-            tool.type != .permissions &&
-            tool.type != .performance
-        }
-
-        let toolsViewController = createToolsController(with: toolItems, viewController: viewController)
-        let deviceViewController = createDeviceViewController()
-
-        tabBarControllers.insert(deviceViewController, at: 0)
-        tabBarControllers.insert(toolsViewController, at: 2)
+    func createTabBarControllers(with tools: [Tool], optionSwitchItems: [OptionSwitchItem], viewController: UIViewController) -> [UIViewController] {
+        var tabBarControllers: [UIViewController] = []
+        tabBarControllers.append(createDeviceViewController())
+        tabBarControllers.append(createApplicationViewController())
+        tabBarControllers.append(createToolsController(with: tools, viewController: viewController))
+        tabBarControllers.append(createPreferencesViewController(optionSwitchItems))
+        tabBarControllers.append(createPerformanceViewController())
         return tabBarControllers
     }
 
     func createToolsController(with tools: [Tool], viewController: UIViewController) -> UIViewController {
-        let navigationItems = tools.map { NavigationToolTableItem(title: $0.name, navigate: $0.presentPreview(from:)) }
+        let navigationItems = tools
+            .map { NavigationToolTableItem(title: $0.name, navigate: $0.presentPreview(from:)) }
         let section = ToolTableSection(title: "Tools", items: navigationItems)
-        let toolsViewController = ToolTable(name: "Tools", sections: [section]).createViewController(on: viewController)
-        toolsViewController.tabBarItem = UITabBarItem(title: "Tools", image: UIImage.Sentinel.tools.resize(), tag: 2)
+        let toolTable = ToolTable(name: "Tools", sections: [section])
+        let toolsViewController = SentinelTableViewController.create(with: toolTable)
+        toolsViewController.tabBarItem = UITabBarItem(
+            title: "Tools",
+            image: UIImage.resize(UIImage.loadImageFromBundle(name: "tools")),
+            tag: 2
+        )
         return toolsViewController
     }
 
     func createDeviceViewController() -> UIViewController {
-        let deviceInfoItem = DeviceTool()
-        return deviceInfoItem.createViewController()
+        let deviceInfoItem = DeviceTool().tool
+        let toolTable = deviceInfoItem.createToolTable(with: deviceInfoItem.info)
+        let viewController = SentinelTableViewController.create(with: toolTable)
+        viewController.tabBarItem = UITabBarItem(
+            title: "Device",
+            image: UIImage.resize(UIImage.loadImageFromBundle(name: "device")),
+            tag: 0
+        )
+        return viewController
+}
+
+    func createApplicationViewController() -> UIViewController {
+        let applicationInfoTool = ApplicationTool().tool
```

Comment:
Same question here
```suggestion
        let applicationInfoTool = ApplicationTool()
```

Reply:
Do you want me to add a function which will use the tool from ApplicationTool or the naming is just wrong? I just removed ".tool" and used it later on if that is what you meant?

Reply:
The tool in application is actually another tool, right?
I was thinking of composition and to hide the existence of that other tool from users of ApplicationTool.
Does that make sense?

Reply:
Yes, it is `CustomInfoTool`. if I understood what you meant, I would create a protocol which would just handle whatever needs to be done with the `Tool`?
 In my case I would need a protocol which a function which returns ToolTable and contains a property `info: [Section].` That way I wouldn't expose the `CustomInfoTool`, just the protocol on which I could call the before mentioned function?

Reply:
You don't need a protocol, you just need to internally (not publically) expose ToolTable from ApplicationTool.